src/tree.ts

```diff
@@ -225,7 +225,10 @@ export class SourceFileNode extends BaseNode {
     const item = new vscode.TreeItem(path.basename(this.filePath));
     item.id = this.id;
     item.resourceUri = vscode.Uri.file(this.filePath);
-    item.contextValue = 'nodeType=file';
+    const name = this.name.toLowerCase();
+    const cml = name == "cmakelists.txt";
+    const compilable = name.search(/\.c/)!==-1;
```

Comment:
I think this should be more explicit. `.c` could exist in a part of the filename that is not the extension.  Consider using `path.extname(name)` to chop off the extension and compare it to the exact extensions we want to support.

Reply:
Which extensions do we want to support?
So far there was no check and the 'compile file' command was shown even for txt files.
The explorer view supports .c and .cpp.

Common c/c++ source file extensions include
- cpp
- c
- cc
- cxx
- c++
according to [wiki](https://en.wikibooks.org/wiki/C%2B%2B_Programming/Programming_Languages/C%2B%2B/Code/File_Organization).

I'd go for the full list and probably also add the extensions to the explorer view.

Reply:
I believe the `resourceLangId` is used for the explorer view so that we don't have to manage the extensions.

I'm also wondering if maybe you can just use the compilation information in the code model instead of file extensions.  This might also cover cases where custom extensions are used.  It looks like the filegroup that contains the source files with have "language" set to "C" or "CXX".  The filegroup with headers will not.  (I just did a quick check with cmake-file-api, I haven't checked with CMake Server to see if it's consistent)

Reply:
Looks like CMake Server and file-api do the same thing.

I added a check for the language. i'll probably do another pull request in the future to clean up the tree.ts file.
i noticed some unused variables and codepaths.

Reply:
Awesome!  Thanks again for doing this!